pulp_rpm/app/advisory.py

```diff
@@ -0,0 +1,206 @@
+from gettext import gettext as _
+from itertools import chain
+
+import hashlib
+
+import createrepo_c as cr
+
+from django.db import (
+    IntegrityError,
+    transaction,
+)
+from django.utils.dateparse import parse_datetime
+
+from pulpcore.plugin.models import (
+    Content,
+    RepositoryContent,
+)
+
+from pulp_rpm.app.exceptions import AdvisoryConflict
+
+
+def resolve_advisories(version):
+    """
+    Decide which advisories to add to a repo version and which to remove, and adjust a repo version.
+
+    Args:
+        version (pulpcore.app.models.RepositoryVersion): current incomplete repository version
+    """
+    content_pks_to_add = set()
+    content_pks_to_remove = set()
+    content_pks_to_exclude = set()  # exclude from the set of content which is being added
+
+    # identify conflicting advisories
+    repo = version.repository
+    base_version = version.base_version
+    previous_version = base_version
+    if not previous_version:
+        repo_version_numbers = sorted(list(repo.versions.values_list("number", flat=True)))
+        # if there are previous versions
+        if len(repo_version_numbers) > 1:
+            previous_version_number = repo_version_numbers[-2]
+            previous_version = repo.versions.get(number=previous_version_number)
```

Comment:
Any chance you can move this to `finalize_new_version` so I can reuse it?

Reply:
This might be a bit cleaner:

```python
if version.base_version:
    previous_version = version.base_version
else:
    previous_version = RepositoryVersion.objects.filter(repository=repo, number__lt=version.number).order_by("-number").first()
```

Reply:
https://github.com/pulp/pulp_rpm/pull/1516/